### ISSUE TITLE
PR4-01: Wire trace-memory consistency wrapper into coarse training

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#459]
# PR4-01: Wire trace-memory consistency wrapper into coarse training

## Goal

Connect the PR3 trace-memory consistency core to the fbpick coarse training path. When `train.trace_memory_consistency.enabled: true`, the training criterion should be wrapped by `TraceMemoryConsistencyCriterion`. When disabled or omitted, training behavior must remain identical to the existing supervised-only criterion.

This PR should not add site54 experiment overlays, OOF experiment configs, or training-loop metric logging. Those remain later PRs.

## Target files

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/train.py`
- Optional import-only updates if needed.

## Required implementation

In `coarse/train.py`, import the wrapper helper from PR3:

```python
from .trace_memory_consistency import maybe_wrap_trace_memory_consistency
```

Replace the direct training criterion assignment in `build_train_bundle()` with a two-step construction:

```python
base_criterion = build_criterion(
    typed.train.pickability_pu,
    coarse_usability=typed.train.coarse_usability,
)
criterion = maybe_wrap_trace_memory_consistency(
    base_criterion=base_criterion,
    cfg=typed.train.trace_memory_consistency,
    fb_sigma_ms=float(typed.train.fb_sigma_ms),
)
```

Keep `criterion_eval = build_criterion()` in `build_train_spec()` unchanged. Trace-memory consistency is training-only and must not affect inner validation / inference loss.

## Expected wrapper order

The intended normal training path is:

```text
TraceMemoryConsistencyCriterion
  -> stock coarse criterion
       -> location KL + optional pickability PU + optional coarse usability BCE
```

For site54 runtime hooks, `maybe_wrap_site54_coarse_train_spec()` wraps `spec.criterion` after `build_train_spec()`. Therefore the expected site54 order becomes:

```text
Site54CoarseExperimentCriterion
  -> TraceMemoryConsistencyCriterion
       -> stock coarse criterion
```

This order is intentional. If site54 velocity masking is enabled, the memory consistency loss receives the velocity-masked coarse pick logits via `pred_for_loss`.

## Constraints

- Do not create the memory bank in a Dataset or DataLoader worker.
- Do not modify eval/infer criterion construction.
- Do not update memory during validation/inference.
- Do not change the PR3 `TraceMemoryConsistencyCriterion` API unless a compile/test failure forces it.
- Do not change default config behavior: omitted `train.trace_memory_consistency` must still be disabled.

## Acceptance criteria

- With `train.trace_memory_consistency.enabled: false`, `bundle.criterion` is the plain criterion returned by `build_criterion()`.
- With `train.trace_memory_consistency.enabled: true`, `bundle.criterion` is a `TraceMemoryConsistencyCriterion`.
- `TraceMemoryConsistencyCriterion.cfg` equals `typed.train.trace_memory_consistency`.
- `TraceMemoryConsistencyCriterion.fb_sigma_ms` equals `typed.train.fb_sigma_ms`.
- `build_train_spec()` still uses an unwrapped eval criterion for `infer_epoch_fn`.
- Existing fbpick coarse tests remain green.
